lib/_stream_writable.js

```diff
@@ -464,9 +464,11 @@ function afterWrite(stream, state, count, cb) {
     stream.emit('drain');
   }
 
+  const err = state.destroyed ? new ERR_STREAM_DESTROYED('write') : undefined;
```

Comment:
attention here, a write should not be able to complete successfully if the stream errored before the write completed, without this a tls test fails.